app/src/hooks/privy/usePrivyLinkEmail.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  useLinkAccount,
+  usePrivy,
+  useUpdateAccount,
+} from "@privy-io/react-auth"
+import { useRef } from "react"
+import { toast } from "sonner"
+
+import { syncPrivyUser } from "@/db/privy"
+
+import { useUser } from "../db/useUser"
+import { useHandlePrivyErrors } from "../useHandlePrivyErrors"
+
+export const usePrivyEmail = (userId: string) => {
+  const isLinking = useRef(false)
+
+  const onError = useHandlePrivyErrors()
+  const { invalidate: invalidateUser } = useUser({ id: userId, enabled: false })
+  const { user: privyUser, unlinkEmail } = usePrivy()
+
+  const { linkEmail } = useLinkAccount({
+    onSuccess: async ({ user: updatedPrivyUser, linkMethod }) => {
+      if (linkMethod === "email" && isLinking.current) {
+        toast.promise(
+          syncPrivyUser(updatedPrivyUser)
```

Comment:
We shouldn't be calling critical actions within the toast. A couple reasons:

1. We shouldnt tie core platform functionality to UI rendering mechanism which can fail for different reasons
2. Error handling is abstracted and can fail